src/react/hooks/paragon/useParagonThemeVariants.js

```diff
@@ -0,0 +1,212 @@
+  }, [themeVariants, currentThemeVariant]);
+
+  useEffect(() => {
+    // Call `onLoad` once both the paragon and brand theme variant are loaded.
+    if (isParagonThemeVariantLoaded && isBrandThemeVariantLoaded) {
+      onLoad();
+    }
+  }, [isParagonThemeVariantLoaded, isBrandThemeVariantLoaded, onLoad]);
+
+  useEffect(() => {
+    if (!themeVariants) {
+      return;
+    }
+
+    /**
+     * Determines the value for the `rel` attribute for a given theme variant based
+     * on if its the currently applied variant.
+     */
+    const generateStylesheetRelAttr = (themeVariant) => (currentThemeVariant === themeVariant ? 'stylesheet' : 'alternate stylesheet');
+
+    // Iterate over each theme variant URL and inject it into the HTML document, if it doesn't already exist.
+    Object.entries(themeVariants).forEach(([themeVariant, value]) => {
+      // If there is no config for the theme variant URL, set the theme variant to loaded and continue.
+      if (!value.urls) {
+        setIsParagonThemeVariantLoaded(true);
+        setIsBrandThemeVariantLoaded(true);
+        return;
+      }
+      const getParagonThemeVariantLink = () => document.head.querySelector(`link[data-paragon-theme-variant='${themeVariant}']`);
+      const existingThemeVariantLink = document.head.querySelector(`link[href='${value.urls.default}']`);
+      const existingThemeVariantBrandLink = document.head.querySelector(`link[href='${value.urls.brandOverride}']`);
+
+      const getExistingThemeVariantLinks = (isBrandOverride) => {
+        const themeVariantLinkSelector = `link[data-${isBrandOverride ? 'brand' : 'paragon'}-theme-variant='${themeVariant}']`;
+        return document.head.querySelectorAll(themeVariantLinkSelector);
+      };
+
+      const createThemeVariantLink = (
+        url,
+        {
+          isFallbackThemeUrl = false,
+          isBrandOverride = false,
+        } = {},
+      ) => {
+        let themeVariantLink = document.createElement('link');
+        themeVariantLink.href = url;
+        themeVariantLink.rel = generateStylesheetRelAttr(themeVariant);
+        if (isBrandOverride) {
+          themeVariantLink.dataset.brandThemeVariant = themeVariant;
+        } else {
+          themeVariantLink.dataset.paragonThemeVariant = themeVariant;
+        }
+
+        themeVariantLink.onload = () => {
+          if (themeVariant === currentThemeVariant) {
+            if (isBrandOverride) {
+              setIsBrandThemeVariantLoaded(true);
+            } else {
+              setIsParagonThemeVariantLoaded(true);
+            }
+          }
+        };
+
+        themeVariantLink.onerror = () => {
+          logError(`Failed to load theme variant (${themeVariant}) CSS from ${value.urls.default}`);
+          if (isFallbackThemeUrl) {
+            logError(`Could not load theme variant (${themeVariant}) CSS from fallback URL. Aborting.`);
+            if (isBrandOverride) {
```

Comment:
Similar question around test coverage on some of the uncovered code paths within this hook.

Reply:
@brian-smith-tcril to double check to see if this has been updated

Reply:
This conversation thread can be resolved (I no longer have permissions to do so myself).